src/ItemFieldSet.tsx

```diff
@@ -1,63 +1,114 @@
+import Button from '@material-ui/core/Button';
 import Card from '@material-ui/core/Card';
 import CardContent from '@material-ui/core/CardContent';
+import Checkbox from '@material-ui/core/Checkbox';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import FormGroup from '@material-ui/core/FormGroup';
+import FormLabel from '@material-ui/core/FormLabel';
+import IconButton from '@material-ui/core/IconButton';
 import TextField from '@material-ui/core/TextField';
+import AddCircleIcon from '@material-ui/icons/AddCircle';
+import { Currencies, Money } from 'ts-money';
 
 import * as React from 'react';
 
 export interface IItem {
+  isFocused: boolean,
   name: string,
   payingIndexes: number[], // Indexes of people who will pay for this item 
-  price?: number,
-  quantity: number,
+  price: string,
+  quantity: string
 }
 
 interface IItemFieldSetProps {
   names: string[],
   setItems: (items: IItem[]) => void,
 }
 interface IItemFieldSetState {
-  items: IItem[]
+  items: IItem[],
 }
 
 export class ItemFieldSet extends React.Component<IItemFieldSetProps, IItemFieldSetState> {
   public state = {
     items: [
       {
+        isFocused: false,
         name: "",
         payingIndexes: [],
-        quantity: 1,
+        price: "",
+        quantity: "1",
       }
     ]
   }
 
   public render() {
-    const names = this.props.names.map((name: string) => {
-      return (<p key={name}>{name}</p>);
+    const names = (itemIndex: number) => this.props.names.map((name: string, nameIndex: number) => {
+      return (<FormControlLabel
+        key={(itemIndex + "-" + nameIndex)}
+        control={
+          <Checkbox checked={this.isPayingForItem(itemIndex, nameIndex)} onChange={this.isPayingForItemNameToggle(itemIndex, nameIndex)} value={name} />
+        }
+        label={name}
+      />
+      )
     });
-    return this.state.items.map((item: IItem, index: number) => {
-      return (
-        <Card key={index}>
+    const splitBillButton = (<Button id="splitBtn" variant="contained" color="primary" onClick={this.setItems} disabled={!this.isValid()}>
+      Split bill
+		</Button>);
+    const addButton = (<IconButton style={{ float: 'right' }} iaria-label="Add " onClick={this.add}>
+      <AddCircleIcon color="secondary" />
+    </IconButton>);
+    const itemsArray = this.state.items.map((item: IItem, itemIndex: number) => {
+      return (<div key={itemIndex}>
+        <Card key={itemIndex}>
           <CardContent>
             <p>Describe Item Consumed</p>
             <div>
               <TextField type="text"
                 label="Product Name"
                 value={item.name}
-                onChange={this.changeItemName(index)}
+                onChange={this.changeItemName(itemIndex)}
               />
             </div>
+            <br />
             <div>
+              <FormLabel component="legend">Quantity</FormLabel>
               <TextField type="text"
```

Comment:
Use type="number" instead, it will show a distinct keyboard on mobile phones that will allow only numbers.